### PR TITLE
fix: Ignore old Vaadin provided dependencies for bundle resolution

### PR DESCRIPTION
**Pull request by mshabarov**
Ignores the dependencies provided by older Vaadin versions and not present anymore in a target Vaadin version. Examples: polymer iron-list or deprecated Vaadin web components with "vaadin-" prefix.

Fixes https://github.com/vaadin/flow/issues/15910

### CODE REVIEW

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskRunDevBundleBuild.java [hunk 1]
```diff
@@ -709,6 +707,48 @@ private void runFrontendBuildTool(String toolName, String executable,
...
+     * erroneously.
+     * <p>
+     * Examples: polymer iron-list dependencies or Vaadin web components with a
+     * deprecated "vaadin-" prefix that are not a part of platform 24.0 anymore.
+     *
+     * @param packageJson
+     *            content of the package.json content red from a file
+     */
+    private static void cleanOldPlatformDependencies(JsonObject packageJson) {
+        if (packageJson == null) {
```
Comment (caalador) [inline flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskRunDevBundleBuild.java:725]:
```suggestion
        if (packageJson == null || !hasFrameworkDependencyObjects(packageJson)) {
```

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskRunDevBundleBuild.java [hunk 2]
```diff
@@ -709,6 +707,48 @@ private void runFrontendBuildTool(String toolName, String executable,
...
+                        .equals(dependencies.getString(vaadinDependency))) {
+                    dependencies.remove(vaadinDependency);
+                    getLogger().debug(
+                            "Old Vaadin provided dependency '{}':'{}' has been removed from package.json",
+                            vaadinDependency, version);
+                }
+            }
+        }
+    }
+
```
Comment (caalador) [inline flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskRunDevBundleBuild.java:751]:
```suggestion

    private static boolean hasFrameworkDependencyObjects(JsonObject packageJson) {
        return packageJson.hasKey(NodeUpdater.VAADIN_DEP_KEY)
                && packageJson.getObject(NodeUpdater.VAADIN_DEP_KEY)
                .hasKey(NodeUpdater.DEPENDENCIES) && packageJson.hasKey(
                NodeUpdater.DEPENDENCIES);
    }

```

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskRunDevBundleBuild.java [hunk 3]
```diff
@@ -709,6 +707,48 @@ private void runFrontendBuildTool(String toolName, String executable,
...
+        JsonObject vaadinDepObject = packageJson
+                .getObject(NodeUpdater.VAADIN_DEP_KEY);
+
+        JsonObject dependencies = packageJson
+                .getObject(NodeUpdater.DEPENDENCIES);
+
+        if (dependencies != null && vaadinDepObject != null
+                && vaadinDepObject.hasKey(NodeUpdater.DEPENDENCIES)) {
+            JsonObject vaadinDependencies = vaadinDepObject
+                    .getObject(NodeUpdater.DEPENDENCIES);
```
Comment (caalador) [inline flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskRunDevBundleBuild.java:738]:
```suggestion
        JsonObject dependencies = packageJson
                .getObject(NodeUpdater.DEPENDENCIES);

        JsonObject vaadinDependencies = vaadinDepObject
                .getObject(NodeUpdater.VAADIN_DEP_KEY)
                .getObject(NodeUpdater.DEPENDENCIES);
```

**Review by caalador — COMMENTED** [review]
Some minor streamlining of the package object checks.

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskRunDevBundleBuild.java (hunk 1 above)
Comment (mshabarov) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskRunDevBundleBuild.java]: Done

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskRunDevBundleBuild.java (hunk 2 above)
Comment (mshabarov) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskRunDevBundleBuild.java:747]: Done

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskRunDevBundleBuild.java (hunk 3 above)
Comment (mshabarov) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskRunDevBundleBuild.java]: Done

**Review by caalador — APPROVED** [review]

**Pull request merged**